src/hermes/commands/deposit/invenio.py

```diff
@@ -132,6 +137,90 @@ def deposit(click_ctx: click.Context, ctx: CodeMetaContext):
     _log.info("Published record: %s", record["links"]["record_html"])
 
 
+def _resolve_latest_invenio_id(ctx: CodeMetaContext) -> str:
+    """
+    Using the given configuration and metadata, figure out the latest record id.
+
+    If a record id is present as configuration ``deposit.invenio.record_id`` this one will be used to identify the
+    latest version of the record. Otherwise, if there is a doi present (either as configuration with key
+    ``deposit.invenio.doi``  or as a codemeta identifier, the DOI will be used to resolve the base record id.
+
+    Anyway, the record id will always be use to resolve the latest version.
+
+    If some of the resolution steps fail or produce an unexpected result, a ValueError will be thrown.
+
+    :param ctx: The context for which the record id should be resolved.
+    :return: The Invenio record id.
+    """
+
+    cfg = config.get('deposit').get('invenio', {})
+    base_url = cfg.get('base_url', 'https://sandbox.zenodo.org')
+    record_id = cfg.get('record_id')
+
+    # Check if we configured an Invenio record ID (of the concept...)
+    if record_id is None:
+        doi = cfg.get('doi')
+        if doi is None:
+            # TODO: There might be more seantic in the codemeta.identifier... (also see schema.org)
+            identifier = ctx['codemeta.identifier']
+            if identifier.startswith('https://doi.org/'):
+                doi = identifier[16:]
```

Comment:
```suggestion
            if identifier.startswith('https://doi.org/'):
                doi = identifier[16:]
            elif identifier.startswith('http://dx.doi.org/'):
                doi = identifier[18:]
```

This catches the other common (older) resolver URL which is still found in abundance. As both may come as HTTP *and* HTTPS URLs, I've created #147 to track this.

As CodeMeta `identifier` can also contain the actual DOI (excluding the resolver URL), these cases should be handled as well. Created #148 for this.